### ISSUE TITLE
Keep Anndata .obs and .var information in batch correction

### ISSUE THREAD

**Issue by christie-ga** [brianhie/scanorama#57]
Hi!

Thank you very much for your great tool, Unfortunately, I am still having some problems with it and I hope you can help me with that. 

I am using the scanorama.correct_scanpy() method for data integration and batch correction on a list of AnnData objects. As a result I get the corrected expression values but all the additional information that was stored in AnnData.obs and AnnData.var is lost. Is this to be expected?

If so, how can I add the corrected expression values to my originial AnnData objects as the gene order is changed after batch correction?

Thank you!

**Comment by brianhie** [comment]
Good to know, this is to be expected since the current version of the code creates new AnnData objects and populates them with only the expression matrix and sorted gene names (https://github.com/brianhie/scanorama/blob/master/scanorama/scanorama.py#L235).

Changing this will require more work on my end (which I currently don't have time for, but if you do want to fix this I'd love to review a pull request!). I'll keep this issue open as well.

For now, you could make sure that you provide Scanorama your genes in sorted order (so it doesn't change them) _and_ that you give Scanorama only the genes in the intersection among all your datasets.

There is also a workaround where you map genes to their respective index and then remap the Scanorama gene order to your original gene order. For example, something like
```python
datasets = [ adata1.X, adata2.X ]
genes_list = [ adata1.var_names, adata2.var_names ]
assert(adata1.var_names == adata2.var_names) # Same genes for both datasets.

corrected, genes = scanorama.correct(datasets, genes_list)
gene2idx = { gene: idx for idx, gene in enumerate(genes) }

adata1.obsm['X_scanorama'] = corrected[0][[ gene2idx[gene] for gene in adata1.var_names ]]
adata2.obsm['X_scanorama'] = corrected[1][[ gene2idx[gene] for gene in adata2.var_names ]]
```

I haven't run this code, but it would be something like that.

**Comment by nahanoo** [comment]
I agree with @christie-ga that it would be awesome to keep the information stored in the uncorrected AnnData.obs.

I'll try your suggested code and post the outcome.

**Comment by nahanoo** [comment]
Would it also be possible to do something like this?

```
integrated = scanorama.integrate_scanpy(adatas)
corrected = scanorama.correct_scanpy(adatas)
integrated, corrected = scanorama.correct_scanpy(adatas, return_dimred=True)
adata_corrected = corrected[0].concatenate(corrected[1:])

adata_corrected.obs['cell_type'] = adata_uncorrected.obs['cell_type'].to_list()
```

Where `adata_uncorrected` are the concatenated uncorrected adatas.

**Comment by brianhie** [comment]
@nahanoo scanorama doesn't change cell order, so it seems like that code would work, but would be good to make sure. I'll mark this as a feature request to do.

**Comment by xguse** [comment]
> @nahanoo scanorama doesn't change cell order

So can you correct me if I am wrong? Since the cell order is not changed we should be able to simply do something like this to get our obs annotations into the corrected adata obj?

```
integrated, corrected = scanorama.correct_scanpy(
        [orig1, orig2],
        return_dimred=True,
        dimred=20,
    )

corrected[0].obs = orig1.obs
corrected[1].obs = orig2.obs

adatas = sc.concat(corrected)
```

And this method brings the original index, which is more helpful than an index of numbers since they usually contain the barcodes we might need to tie data between cells across different analysis streams.

In addition to this, can we copy over the NON-var related data from the original adata objs like `uns` etc?

I am very grateful for this integration tool but the lack of anything more than a cursory attempt at documentation and the fact that I lose ALL of my metadata makes it almost unusable in any context other than playing around.

**Comment by brianhie** [comment]
@xguse thanks for the feedback -- the issue is that Scanorama changes gene order, see https://github.com/brianhie/scanorama/pull/71. Let me get back to you on this issue.

**Comment by brianhie** [comment]
Issue addressed in https://github.com/brianhie/scanorama/pull/77 and released in Scanorama v1.7. Note the API change (see README and doc string).
